primitives/runtime/src/chain.rs

```diff
@@ -85,3 +85,6 @@ pub type HasherOf<C> = <C as Chain>::Hasher;
 
 /// Header type used by the chain.
 pub type HeaderOf<C> = <C as Chain>::Header;
+
+/// Number type used in the Header type used by the chain
+pub type HeaderNumberOf<C> = <<C as Chain>::Header as HeaderT>::Number;
```

Comment:
You could actually use `BlockNumerOf<>` instead of this, because `Chain::Header` is defined as `HeaderT<Number = Self::BlockNumber>` :)